lib/modules/manager/devbox/schema.ts

```diff
@@ -0,0 +1,32 @@
+import { z } from 'zod';
+import { Jsonc } from '../../../util/schema-utils';
+
+type Packages = Record<string, string | undefined>;
+
+export const DevboxSchema = z.object({
+  packages: z
+    .union([
+      z.array(z.string()),
+      z.record(z.union([z.string(), z.object({ version: z.string() })])),
+    ])
+    .transform((packages): Packages => {
+      const result: Packages = {};
+      if (Array.isArray(packages)) {
+        for (const pkg of packages) {
+          const [name, version] = pkg.split('@');
+          result[name] = version;
+        }
+      } else {
+        for (const [name, value] of Object.entries(packages)) {
+          if (typeof value === 'string') {
+            result[name] = value;
+          } else {
+            result[name] = value.version;
+          }
```

Comment:
Example of unifying transform:

```ts
z.union([
  z.string(),
  z.object({ version: z.string() }).transform(({ version }) => version),
])
```

This way, you don't need to switch based on `typeof` anymore, as you have same type from both branches.